test/functional/core/remote_spec.lua

```diff
@@ -0,0 +1,60 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local eq = helpers.eq
+local neq = helpers.neq
+local funcs = helpers.funcs
+local nvim_prog_abs = helpers.nvim_prog_abs
+local merge_args = helpers.merge_args
+local spawn = helpers.spawn
+local set_session = helpers.set_session
+local expect = helpers.expect
+
+describe("remote --server client", function()
+  local server, client, address, pid
+  before_each(function()
+    local nvim_argv = merge_args(helpers.nvim_argv, {'--headless'})
+    server = spawn(nvim_argv)
+    set_session(server)
```

Comment:
I think `merge_args` can be "private", tests don't need to call it directly anymore. `spawn()` and `clear()` merge args implicitly. 

    server = spawn({'--headless'}, true)

Reply:
`clear()` is killing the previous session though. But we could add it an extra flag for that as well.